src/Database/Manager.js

```diff
@@ -50,21 +50,27 @@ class DatabaseManager {
 
   /**
    * Creates a new database connection for the config defined inside
-   * `config/database` file. You just need to pass the key name or don't
-   * pass any name to use the default connection.
+   * `config/database` file or from a provided JSON object. You just
+   * need to pass the key name or don't pass any name to use the default
+   * connection. For JSON objects your object should have a key 'name'
+   * that defines the pool.
+   *
    *
    * Also this method will reuse and returns the existing connections.
    *
    * @method connection
    *
-   * @param  {String}   [name = Config.get('database.connection')]
+   * @param  {String | Object}   [name = Config.get('database.connection')]
    *
    * @return {Database}
    *
-   * @throws {missingDatabaseConnection} If connection is not defined in config file.
+   * @throws {missingDatabaseConnection} If a string is provided and the connection is not defined in config file.
    */
-  connection (name) {
-    name = name || this.Config.get('database.connection')
+  connection (conn) {
+    conn = conn || this.Config.get('database.connection')
+
+    let name = typeof conn === 'string' ? conn : conn['name']
```

Comment:
Can you please explain this line?

Reply:
Are you referring to line 72, line 70 or both?

Line 70 was simply a variable rename for the default lookup for the connection in the case that "conn" was not provided.

Line 72 gets the pool name either from the provided JSON object or from the string provided as a parameter so that it can be used for creating a pool. This causes a small divergence from the config/database.js JSON objects as they can be identified in the pool by their 'database.json_object_name' identifier. If I'm incorrect on the need for a name in order for pooling to work let me know.